lib/web/integrations_awsoidc.go

```diff
@@ -427,6 +429,50 @@ func (h *Handler) awsOIDCConfigureEKSIAM(w http.ResponseWriter, r *http.Request,
 		return nil, trace.BadParameter("invalid role %q", role)
 	}
 
+	var (
+		integrationRoles = make(map[string]struct{})
+		integrations     []types.Integration
+		nextPage         string
+		err              error
+	)
+	// Ensure the passed role exists in an AWS Integration resource. If
+	// it doesn't an attacker could use the created script to give
+	// themselves the permissions that should be granted to an
+	// integration role instead.
+	for {
+		integrations, nextPage, err = h.cfg.AccessPoint.ListIntegrations(r.Context(), 0, nextPage)
```

Comment:
This does have the downside of querying all integrations every time this endpoint is hit, I could also dynamically update a map that keeps track of unique AWS roles used in integrations if this is deemed unacceptable. I figured I'd take the simplest approach first and change it if I need to.